components-core/src/main/java/com/adyen/checkout/components/ActionHandlingComponent.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 11/11/2022.
+ */
+
+package com.adyen.checkout.components
+
+import android.app.Activity
+import android.content.Intent
+import androidx.annotation.RestrictTo
+import androidx.lifecycle.LifecycleOwner
+import com.adyen.checkout.components.model.payments.response.Action
+import kotlinx.coroutines.CoroutineScope
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+interface ActionHandlingComponent {
+
+    fun handleAction(action: Action, activity: Activity)
+
+    fun handleIntent(intent: Intent)
+
+    fun observeAction(
```

Comment:
I think this should not be its own method, `PaymentComponentEvent` should include a new event for the action and the merchant would only need one `observe`. WDYT?

Reply:
Good suggestion 👍